.github/workflows/benchmarks.yml

```diff
@@ -5,10 +5,6 @@ on:
     branches: [main]
     paths-ignore:
       - '**/*.md'
-  pull_request:
```

Comment:
i thought we did want benchmarking per PR though

Reply:
how do you feel about it? I just polled inside the office. Kinda felt like `on push` is likely fine. That means if we really have done something bad with the PR in terms of benchmarks, then we catch it after the fact.

Reply:
I guess more because of the mac runner situation.

Reply:
i'm okay with it only doing it on push; at least we still will have a linear benchmark history on the `main` branch that we can pinpoint to a particular merge that regressed.

we probably do NOT need multiple data points within the same PR for each push.

Reply:
> probably do NOT need multiple data points
want to clarify: are you referring to the python/os matrix or you're agreeing the reduction in this PR is fine?

Reply:
i agree with the reduction here. we don't need to run for each PR